### DISCUSSION TITLE
Adding custom external login options

### DISCUSSION THREAD

**Discussion by jesseg34** [directus/directus#4865] [Archive]
Does directus 9 offer the ability to use an external auth provider that is not predefined in the grant lib? I want to leverage Identity Server 4 to do this but am not seeing how to make that happen in the docs. Thanks in advance!

**Comment by rijkvanzanten** [comment]
@jesseg34 Grant is a generic oAuth abstraction layer, so as long as your service supports oAuth2, it'll work with grant (and thus Directus) 🙂 

The only thing grant does for the listed providers on the repo is that they have some of the configuration options pre-filled, so you don't have to configure _everything_ (see https://github.com/simov/grant/blob/master/config/oauth.json and https://github.com/simov/grant/blob/master/config/profile.json for example).
